### ISSUE TITLE
Nodejs 8 (node:4376) [DEP0062] DeprecationWarning: node --debug and node --debug-brk are invalid. Please use node --inspect or node --inspect-brk instead. Was: - Cannot debug JavaScript via Node.JS (Failed to connect to Standalone V8 VM)- Cannot debug JavaScript via Node.JS (Failed to connect to Standalone V8 VM)

### ISSUE THREAD

**Issue by paulvi** [Nodeclipse/nodeclipse-1#220]
Initially raised as https://github.com/Nodeclipse/www.nodeclipse.org/issues/77

It seems that Nodejs 8 change option names

> (node:4376) [DEP0062] DeprecationWarning: `node --debug` and `node --debug-brk` are invalid. Please use `node --inspect` or `node --inspect-brk` instead.

The line in question is https://github.com/Nodeclipse/nodeclipse-1/blob/master/org.nodeclipse.debug/src/org/nodeclipse/debug/launch/LaunchConfigurationDelegate.java#L114

> cmdLine.add("--debug"+brk+"="+nodeDebugPort);

i.e. using `--debug` is hard-coded now

TODO test with Node.js 8, if using `--inspect` would work.

HELP WANTED!!

**Comment by dmidge8** [comment]
I still have the same issue, that I found through issue #107 .
"Nodeclipse/chromedevtools failed to connect to Standalone V8 VM ( Check Help (F1) and Support http://www.nodeclipse.org/#support )
. Info:
Connexion refusée (Connection refused)"

I am using nodeclipse 1.0.2, which seems to be the latest version. Is the fix included in this version?

**Comment by 1fifoto** [comment]
I could really use this. Is there anything I can do to help?

**Comment by paulvi** [comment]
yes, change https://github.com/Nodeclipse/nodeclipse-1/blob/master/org.nodeclipse.debug/src/org/nodeclipse/debug/launch/LaunchConfigurationDelegate.java#L114

build, try that everything works fine

then I will release or help to release

**Comment by paulvi** [comment]
Also need to know at what Node.js version this option where renamed,
and do logic to support newer and older versions

**Comment by tyllastarr** [comment]
I am having the issue with Node.js v8.11.3.  Haven't tested with Node.js 10 yet, but I plan to later.

**Comment by Fr33dan** [comment]
Unfortunately it's not a renamed option but a completely new replacement. According to the [Node.js V6 change log](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V6.md#6.3.0) the `--inspect` option was added in 6.3.0 to use the "inspector protocol" as an experimental feature which is different than the debugger protocol. Then the [V8 changelog](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V8.md#8.0.0) says:
  * The `--debug` command-line argument has been deprecated. Note that
    using `--debug` will enable the *new* Inspector-based debug protocol
    as the legacy Debugger protocol previously used by Node.js has been
    removed. [[`010f864426`](https://github.com/nodejs/node/commit/010f864426)]
    [#12949](https://github.com/nodejs/node/pull/12949).

So the issue is not that the the VM failed to launch because of the changed option, but that the debugger can't connect to the VM because the protocol has changed.

I did go ahead and changing the option in LaunchConfigurationDelegate.java which made the depreciation warning go away, but it still thinks the VM failed to launch.

**Comment by gregfenton** [comment]
So I'm guessing the continuation of this issue when running Node-v10.15.2 is expected, yet not desired?
Windows 10, Node-v10, Nodeclipse   Enide/Nodeclipse 1.0.2.201509250223.

Is the only solution to downgrade Node.JS to pre-V8?

**Comment by paulvi** [comment]
I am not using Node for few years,  so don't have real reason to follow node version changes.

Some one should step in, willing to solve the issue with Java code.  
I am gonna help to answer any question on how to release.

HELP WANTED.

**Comment by david-loran** [comment]
paulvi, it's been many years since I did Eclipse plugin development, so I'm not sure I can help.  But isn't it just a matter of changing the nodeclipse plugin's launch configuration of the Node app (e.g., /usr/local/bin/node) to use --inspect or --inspect-brk instead of the deprecated --debug or --debug-brk parameter?

**Comment by david-loran** [comment]
Sorry, just seeing Fr33dan's comment re. change in protocol above.

**Comment by paulvi** [comment]
Mostly time is need for testing this change,  
 then doing release.

I can help with the last.  

I don't expect someone is going to take over this project.  Otherwise I would explain and help to do release by other person.

**Comment by Derek-Ashmore** [comment]
Over a year and still no progress on this issue -- Wow!

**Comment by paulvi** [comment]
It will not be, until someone steps in to actually do it.

I am not using Node.js any more, so not going to follow what changes with every version of Node

**Comment by mrk1989** [comment]
Please fix this command for replace --debug to --inspect.

The comment https://github.com/Nodeclipse/nodeclipse-1/issues/220#issuecomment-412632853 is correct!

C:\Users\muletto10>npm -v
6.14.11

C:\Users\muletto10>node -v
v14.16.0

C:\Users\muletto10>java -version
java version "1.8.0_251"
Java(TM) SE Runtime Environment (build 1.8.0_251-b08)
Java HotSpot(TM) Client VM (build 25.251-b08, mixed mode)

C:\Users\muletto10>node.exe --inspect-brk=1337 C:/Users/muletto10/eclipse-workspace-VA/first/hello-world-server.js
Debugger listening on ws://127.0.0.1:1337/fe46b94c-fe96-47c5-8c2e-3df106065b48
For help, see: https://nodejs.org/en/docs/inspector
^C

**Comment by mrk1989** [comment]
Well , if I do "run Node Program" it work
